src/axom/core/Array.hpp

```diff
@@ -792,9 +799,27 @@ Array<T, DIM, SPACE>::Array(Array&& other)
 //------------------------------------------------------------------------------
 template <typename T, int DIM, MemorySpace SPACE>
 template <typename OtherArrayType>
-Array<T, DIM, SPACE>::Array(const ArrayBase<T, DIM, OtherArrayType>& other)
+Array<T, DIM, SPACE>::Array(const ArrayBase<T, DIM, OtherArrayType>& other,
+                            int allocatorId)
   : ArrayBase<T, DIM, Array<T, DIM, SPACE>>(other)
-  , m_allocator_id(axom::detail::getAllocatorID<SPACE>())
+  , m_allocator_id(allocatorId)
+{
+  initialize(static_cast<const OtherArrayType&>(other).size(),
+             static_cast<const OtherArrayType&>(other).size());
+  // axom::copy is aware of pointers registered in Umpire, so this will handle
```

Comment:
Minor:  This comment might be a little misleading since `axom::copy` also works in configs that don't have Umpire.